modules/sfp_opencorporates.py

```diff
@@ -0,0 +1,131 @@
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["AFFILIATE_COMPANY_NAME", "GEOINFO"]
+
+    # Query the REST API
+    # https://api.opencorporates.com/documentation/API-Reference
+    def query(self, qry):
+        params = {
+            'q': qry,
+            'format': 'json',
+            'order': 'score',
+            'confidence': str(self.opts['confidence'])
+        }
+        res = self.sf.fetchUrl("https://api.opencorporates.com/v0.4/companies/search?" + urllib.urlencode(params),
+                               timeout=self.opts['_fetchtimeout'], useragent=self.opts['_useragent'])
+
+        return res['content']
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+        else:
+            self.results[eventData] = True
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        # Query the API for company information
+        res = self.query(eventData + "*")
+
+        if res is None:
+            return None
+
+        # Parse response content as JSON
+        try:
+            data = json.loads(res)
+        except Exception as e:
+            self.sf.debug("Error processing JSON response: " + str(e))
+            return None
+
+        if 'results' not in data:
+            self.sf.debug("Found no results for " + eventData)
+            return None
+
+        if 'companies' not in data['results']:
+            self.sf.debug("Found no results for " + eventData)
+            return None
+
+        # Check for a match
+        for company in data['results']['companies']:
+            if not eventData.lower() == company['company']['name'].lower():
+                continue
+
+            # Extract registered address
+            location = company['company']['registered_address_in_full']
+
+            if location is not None:
+                if len(location) < 3 or len(location) > 100:
+                    self.sf.debug("Skipping likely invalid location.")
+                else:
+                    if company['company']['registered_address'] is not None:
+                        if company['company']['registered_address']['country'] is not None:
+                            country = company['company']['registered_address']['country']
+                            if not location.endswith(country):
+                                location = location + ", " + country
+
+                    self.sf.info("Found company address: " + location)
+                    e = SpiderFootEvent("GEOINFO", location, self.__name__, event)
+                    self.notifyListeners(e)
+
+            # Extract previous company names
+            previous_names = company['company']['previous_names']
```

Comment:
A previous name isn't an affiliate though?

Reply:
True, but that seemed to fit with the existing paradigm. Similar to how the `AFFILIATE_` prefix is used elsewhere.

I also wasn't sure if you would be keen to have this module report previous names at all. For example, no modules other than the `sfp_names` module are permitted to return a `HUMAN_NAME`. I wasn't sure if the same applied to company names.

Reply:
`RAW_RIR_DATA` instead, perhaps?

Reply:
Updated to use `COMPANY_NAME`.